kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseRuntimeDataServiceIntegrationTest.java

```diff
@@ -1411,6 +1416,115 @@ public void testGetCaseTasksAsStakeholder() throws Exception {
+
+        List<TaskSummary> tasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseClaimId, USER_YODA, 0, 10);
+        assertNotNull(tasks);
+        int countOfTaskBefore = tasks.size();
+
+        assertNotNull(stage.getIdentifier());
+        caseClient.triggerAdHocFragmentInStage(CONTAINER_ID, caseClaimId, stage.getIdentifier(), SUBMIT_POLICE_REPORT_TASK, Collections.EMPTY_MAP);
+
+        tasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseClaimId, USER_YODA, 0, 10);
+        assertNotNull(tasks);
+        assertEquals(countOfTaskBefore + 1, tasks.size());
+
+        TaskSummary task = tasks.get(0);
+        assertNotNull(task);
+        assertEquals(SUBMIT_POLICE_REPORT_TASK, task.getName());
+
+        caseClient.putCaseInstanceData(CONTAINER_ID, caseClaimId, "claimReportDone", Boolean.TRUE);
+        stages = caseClient.getStages(CONTAINER_ID, caseClaimId, false, 0, 10);
+        assertEquals(3, stages.size());
+        assertBuildClaimReportCaseStage(stages.get(0), "Completed");
+        assertClaimAssesmentCaseStage(stages.get(1), "Active");
+        assertEscalateRejectedClaimCaseStage(stages.get(2), "Available");
+
+        tasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseClaimId, USER_YODA, 0, 10);
+        assertNotNull(tasks);
+        assertEquals(1, tasks.size());
+        task = tasks.get(0);
+        assertNotEquals(SUBMIT_POLICE_REPORT_TASK, task.getName());
+
+        try {
+            caseClient.triggerAdHocFragmentInStage(CONTAINER_ID, caseClaimId, stage.getIdentifier(), SUBMIT_POLICE_REPORT_TASK, Collections.EMPTY_MAP);
+            fail("Task was triggered to completed stage");
+        } catch (KieServicesHttpException e) {
+            assertEquals(Integer.valueOf(400), e.getHttpCode());
+        } catch (KieServicesException e) {
+            //ok
+        }
+
+        caseClient.destroyCaseInstance(CONTAINER_ID, caseClaimId);
+    }
+
+    @Test
+    public void testTriggerTaskInCanceledCase() throws Exception {
+        String caseId = startUserTaskCase(USER_YODA, USER_JOHN);
+
+        assertNotNull(caseId);
+        assertTrue(caseId.startsWith(CASE_HR_ID_PREFIX));
+
+        List<TaskSummary> caseTasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseId, USER_YODA, 0, 10);
+        assertNotNull(caseTasks);
+        assertEquals(1, caseTasks.size());
+
+        caseClient.cancelCaseInstance(CONTAINER_ID, caseId);
+
+        try {
+            caseClient.triggerAdHocFragment(CONTAINER_ID, caseId, HELLO_2_TASK, Collections.EMPTY_MAP);
+            fail("AdHoc Task was trigged from canceled case.");
+        } catch (KieServicesHttpException ex) {
+            assertTrue(ex.getMessage().startsWith("Unexpected HTTP response code"));
+            assertEquals(Integer.valueOf(404), ex.getHttpCode());
+        } catch (KieServicesException e) {
+            assertEquals("Case with id " + caseId + " was not found", e.getMessage());
```

Comment:
It is not obvious which exception should be thrown here since there are multiple catch statements. What is the expected behavior?